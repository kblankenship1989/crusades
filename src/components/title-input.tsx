import React from 'react';
import {Input, Icon} from 'react-native-elements';
import {getColorScheme} from '../helpers/getColorScheme';
import {appStyles} from '../../styles';

type TitleInputProps = {
    value: string,
    placeholder?: string,
    onChangeText: (title: string) => void
}
const colorScheme = getColorScheme();
const styles = appStyles(colorScheme);


export const TitleInput = (props : TitleInputProps) : JSX.Element => (
    <Input
        leftIcon={
            <Icon
                size={18}
                name={'format-title'}
                type={'material-community'}
            />
        }
        placeholder={props.placeholder || 'Title'}
        onChangeText={props.onChangeText}
        value={props.value}
        style={styles.textInput}
    />
);
