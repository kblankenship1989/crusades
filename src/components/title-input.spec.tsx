import React from 'react';
import {render} from '@testing-library/react-native';

import {TitleInput, TitleInputProps} from './title-input';


describe('Given the Title Input component', () => {
    describe('And a placeholder value is provided', () => {
        it('should render an input with the provided placeholder text', async () => {
            const testProps : TitleInputProps = {
                placeholder: 'Some placeholder text',
                value: 'Some title',
                onChangeText: jest.fn()
            };
            const component = (<TitleInput
                {...testProps}
            />);
            const {
                queryByPlaceholderText
            } = render(component);

            expect(await queryByPlaceholderText(testProps.placeholder)).toBeTruthy();
        });
    });

    it('should render the component correctly', () => {
        const testProps : TitleInputProps = {
            placeholder: 'Some placeholder text',
            value: 'Some title',
            onChangeText: jest.fn()
        };
        const component = (<TitleInput
            {...testProps}
        />);
        expect(render(component).toJSON()).toMatchSnapshot();
    });
});
