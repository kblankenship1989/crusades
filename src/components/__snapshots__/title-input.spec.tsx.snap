// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Title Input component should render the component correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#86939e",
        "flexDirection": "row",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginVertical": 4,
          "paddingRight": 4,
        }
      }
    >
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 18,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              ï—´
            </Text>
          </View>
        </View>
      </View>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      onChangeText={[MockFunction]}
      placeholder="Some placeholder text"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "color": "black",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      testID="RNE__Input__text-input"
      underlineColorAndroid="transparent"
      value="Some title"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  />
</View>
`;
